### PR TITLE
Use ROS_DISTRO rather than UBUNTU_DISTRO to decide connext version.

### PR DESCRIPTION
**Pull request by nuclearsandwich**
This both better matches what we recommend and matches the support which was just added for RHEL.

### CODE REVIEW

File: linux_docker_resources/Dockerfile [hunk 1]
```diff
@@ -89,18 +89,23 @@ RUN if test ${PLATFORM} = x86; then apt-get update && apt-get install --no-insta
 RUN apt-get update && apt-get install --no-install-recommends -y python3-pexpect
 
 # Get and install the RTI web binaries.
-# Connext 5.3.1 for focal and earlier.
-RUN if test ${UBUNTU_DISTRO} != jammy; then cd /tmp && curl --silent https://s3.amazonaws.com/RTI/Bundles/5.3.1/Evaluation/rti_connext_dds_secure-5.3.1-eval-x64Linux3gcc5.4.0.tar.gz | tar -xz; fi
-RUN if test ${UBUNTU_DISTRO} != jammy; then cd /tmp && tar -xvf /tmp/openssl-1.0.2n-target-x64Linux3gcc5.4.0.tar.gz; fi
+# Connext 5.3.1 for Galactic and earlier.
+RUN if [ "$ROS_DISTRO" != "galactic" -o "$ROS_DISTRO" != "foxy" ]; then \
```
Comment (clalancette) [inline linux_docker_resources/Dockerfile:93]: Isn't this logic inverted?  Shouldn't this be `$ROS_DISTRO = "foxy" -o "$ROS_DISTRO = "galactic"` ?

File: linux_docker_resources/Dockerfile [hunk 2]
```diff
@@ -89,18 +89,23 @@ RUN if test ${PLATFORM} = x86; then apt-get update && apt-get install --no-insta
...
 # Connext 6.0.1 for jammy, the evaluation bundles don't contain security extensions so we need to distribute the pro binaries to ourselves.
 COPY rticonnextdds-src/ /tmp/rticonnextdds-src
-RUN for splitpkg in \
-  /tmp/rticonnextdds-src/rti_connext_dds-6.0.1-pro-host-x64Linux.run \
-  /tmp/rticonnextdds-src/rti_connext_dds-6.0.1.25-pro-host-x64Linux.rtipkg \
-  /tmp/rticonnextdds-src/rti_connext_dds-6.0.1.25-pro-target-x64Linux4gcc7.3.0.rtipkg; do \
-    cat $(echo ${splitpkg}.0?? | sort) > $splitpkg; \
-  done
-RUN chmod 755 /tmp/rticonnextdds-src/rti_connext_dds-6.0.1-pro-host-x64Linux.run
+RUN if [ "$ROS_DISTRO" != "galactic" -o "$ROS_DISTRO" != "foxy" ]; then \
```
Comment (clalancette) [inline linux_docker_resources/Dockerfile:99]: Also, shouldn't this be `$ROS_DISTRO != "foxy" -a $ROS_DISTRO != "galactic"`?

**Review by clalancette — CHANGES_REQUESTED** [review]

File: linux_docker_resources/Dockerfile (hunk 1 above)
Comment (nuclearsandwich) [reply linux_docker_resources/Dockerfile:93]: You are correct. These are what happens when you write conditionals late at night.
```suggestion
RUN if [ "$ROS_DISTRO" = "galactic" -o "$ROS_DISTRO" = "foxy" ]; then \
```

File: linux_docker_resources/Dockerfile (hunk 2 above)
Comment (nuclearsandwich) [reply linux_docker_resources/Dockerfile:99]:
```suggestion
RUN if [ "$ROS_DISTRO" != "galactic" -a "$ROS_DISTRO" != "foxy" ]; then \
```

**Review by clalancette — APPROVED** [review]
Looks good to me with green CI on Rolling and at least one of Galactic or Foxy.

**Comment by clalancette** [comment]
We removed Connext 5 support when we removed Foxy support, so closing this one out.

**Pull request closed without merging**